### PR TITLE
Extend custom Media Type rule

### LINKED ISSUE #583: Replace custom Media type with "application/json"
The [MUST use media type versioning [114]](https://opensource.zalando.com/restful-api-guidelines/index.html#114) rule provides an example with a custom media type. Replace the custom media type with a standard media type as described in [Rule 172](https://opensource.zalando.com/restful-api-guidelines/index.html#172).

### PR DESCRIPTION
**Pull request by vadeg**
Replace custom Media Type in Rule 114 examples with `application/json` media type.

Fixes #583

### CODE REVIEW

**Comment by tfrauenstein** [comment]
👍

**Comment by vadeg** [comment]
I reverted back custom media type and updated formatting and wording.

File: chapters/compatibility.adoc [hunk 1]
```diff
@@ -237,16 +237,18 @@ we strongly recommend to only use the first two approaches.
 
 However, when API versioning is unavoidable, you have to design your
 multi-version RESTful APIs using media type versioning (instead of URI
-versioning, see below). Media type versioning is less tightly coupled since
+versioning, see <<115>>). Media type versioning is less tightly coupled since
```
Comment (tfrauenstein) [inline chapters/compatibility.adoc:240]: "instead of URI versioning, see <<115>"  --> "see <<115>"

**Comment by tfrauenstein** [comment]
looks good -- just one editiorial detail. 
Would be great, if you can also add your proposal for more custom type standardization https://github.com/zalando/restful-api-guidelines/issues/588 
Thank you.

**Comment by vadeg** [comment]
@tfrauenstein I have update rule with a custom media type format definition. It is debatable.

**Comment by ePaul** [comment]
:+1:

**Comment by tkrop** [comment]
:+1:

**Pull request merged**